discovery/gossiper.go

```diff
@@ -1261,13 +1279,20 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []n
 		}
 
 		features := lnwire.NewFeatureVector(msg.Features, lnwire.GlobalFeatures)
+		sig, err := msg.Signature.ToSignature()
```

Comment:
Is this needed? The node announcement is already validated, which should make the signature always valid at this point.

Reply:
Is what needed? We're just converting the wire signature format to the `*btcec.Signature`. I actually added a new `ToSignatureBytes`, so we can avoid the intermediate conversion. Replaced in the latest diff!

Reply:
Diff is gone, but think I meant to convert Sig->Signature->SigBytes. But newest diff looks good!